pkg/machine/ignition.go

```diff
@@ -166,24 +168,57 @@ Environment=SYSTEMD_CONF=/etc/systemd/system.conf.d/default-env.conf
 Environment=ENVD_CONF=/etc/environment.d/default-env.conf
 Environment=PROFILE_CONF=/etc/profile.d/default-env.sh
 ExecStart=/usr/bin/bash -c '/usr/bin/test -f ${FWCFGRAW} &&\
-	echo "[Manager]\n#Got from QEMU FW_CFG\nDefaultEnvironment=$(/usr/bin/base64 -d ${FWCFGRAW} | sed -e "s+|+ +g")\n" > ${SYSTEMD_CONF} ||\
-	echo "[Manager]\n#Got nothing from QEMU FW_CFG\n#DefaultEnvironment=\n" > ${SYSTEMD_CONF}'
+        echo "[Manager]\n#Got from QEMU FW_CFG\nDefaultEnvironment=$(/usr/bin/base64 -d ${FWCFGRAW} | sed -e "s+|+ +g")\n" > ${SYSTEMD_CONF} ||\
+        echo "[Manager]\n#Got nothing from QEMU FW_CFG\n#DefaultEnvironment=\n" > ${SYSTEMD_CONF}'
 ExecStart=/usr/bin/bash -c '/usr/bin/test -f ${FWCFGRAW} && (\
-	echo "#Got from QEMU FW_CFG"> ${ENVD_CONF};\
-	IFS="|";\
-	for iprxy in $(/usr/bin/base64 -d ${FWCFGRAW}); do\
-		echo "$iprxy" >> ${ENVD_CONF}; done ) || \
-	echo "#Got nothing from QEMU FW_CFG"> ${ENVD_CONF}'
+        echo "#Got from QEMU FW_CFG"> ${ENVD_CONF};\
+        IFS="|";\
+        for iprxy in $(/usr/bin/base64 -d ${FWCFGRAW}); do\
+                echo "$iprxy" >> ${ENVD_CONF}; done ) || \
+        echo "#Got nothing from QEMU FW_CFG"> ${ENVD_CONF}'
 ExecStart=/usr/bin/bash -c '/usr/bin/test -f ${FWCFGRAW} && (\
-	echo "#Got from QEMU FW_CFG"> ${PROFILE_CONF};\
-	IFS="|";\
-	for iprxy in $(/usr/bin/base64 -d ${FWCFGRAW}); do\
-		echo "export $iprxy" >> ${PROFILE_CONF}; done ) || \
-	echo "#Got nothing from QEMU FW_CFG"> ${PROFILE_CONF}'
+        echo "#Got from QEMU FW_CFG"> ${PROFILE_CONF};\
+        IFS="|";\
+        for iprxy in $(/usr/bin/base64 -d ${FWCFGRAW}); do\
+                echo "export $iprxy" >> ${PROFILE_CONF}; done ) || \
+        echo "#Got nothing from QEMU FW_CFG"> ${PROFILE_CONF}'
 ExecStartPost=/usr/bin/systemctl daemon-reload
 [Install]
 WantedBy=sysinit.target
 `
+	qemuStatic := `[Unit]
+Description=Layer qemu-user-static & Co with rpm-ostree
+Wants=network-online.target
+After=network-online.target
+After=remove-moby.service
+# We run before 'zincati.service' to avoid conflicting with rpm-ostree
+# transactions.
+Before=zincati.service
+ConditionPathIsDirectory=|!/lib/binfmt.d
+ConditionDirectoryNotEmpty=|!/lib/binfmt.d
+
+[Service]
+Type=oneshot
+RemainAfterExit=yes
+# '--allow-inactive' ensures that rpm-ostree does not return an error
+# if the package is already installed. This is useful if the package is
+# added to the root image in a future Fedora CoreOS release as it will
+# prevent the service from failing.
+ExecStart=/usr/bin/rpm-ostree install --apply-live --allow-inactive qemu qemu-user-static qemu-user-binfmt
```

Comment:
@dustymabe what happens with ^^ when an FCOS update is downloaded and applied on a reboot?

Reply:
I'll have to wait until an FCOS update is out, but my semi-educated _guess_ is that `/lib/binfmt.d` will become empty following an  FCOS update, so the installation will be triggered following an update, due to the `Condition*` clauses in the `[Install]` section.   I _can_ report that this _does_ only run once for a given FCOS version, due to those same `Condition*` clauses (once the packages are installed `/lib/binfmt.d` is no longer emtpy).

Reply:
An update should be handled without an issue. You want to make sure this only runs once. [Our docs](https://docs.fedoraproject.org/en-US/fedora-coreos/os-extensions/#_example_layering_vim_and_setting_it_as_the_default_editor)  use `ConditionPathExists=!/var/lib/%N.stamp` and `ExecStart=/bin/touch /var/lib/%N.stamp` for this.

Reply:
@dustymabe when this runs during boot up, does the entire boot sequence pause and wait for the install to complete?  Or can the user begin interacting with the FCOS VM while the install occurs?

Reply:
Should be able to go ahead and start interacting with the machine.

Reply:
+1  I have set up FCOS VMs with a similar `systemd` unit "by hand", and I am able to ssh to the machine, and interact with it while the installation is running.  Once I ssh to the machine, I can use `systemctl status install-qemu-static.service` to check on how the installation is progressing.  (Obviously, I need to wait for the installation to finish before I can run "foreign" binaries).